packages/lib-classifier/src/store/SubjectStore.spec.js

```diff
@@ -37,35 +37,45 @@ describe('Model > SubjectStore', function () {
     return store
   }
   describe('Actions > advance', function () {
-    let rootStore
-    before(function () {
-      rootStore = setupStores()
-    })
-
-    it('should make the next subject in the queue active when calling `advance()`', function () {
-      expect(rootStore.subjects.active.id).to.equal(rootStore.subjects.resources.values().next().value.id)
-      rootStore.subjects.advance()
-      expect(rootStore.subjects.active.id).to.equal(rootStore.subjects.resources.values().next().value.id)
-      expect(rootStore.subjects.resources.get('1')).to.be.undefined()
-    })
-
-    describe('with less than three subjects in the queue', function () {
+    describe('with a full queue', function () {
       let rootStore
-
       before(function () {
         rootStore = setupStores()
       })
 
-      after(function () {
-        rootStore.subjects.populateQueue.restore()
+      it('should make the next subject in the queue active', function () {
+        expect(rootStore.subjects.active.id).to.equal(rootStore.subjects.resources.values().next().value.id)
+        rootStore.subjects.advance()
+        expect(rootStore.subjects.active.id).to.equal(rootStore.subjects.resources.values().next().value.id)
+        expect(rootStore.subjects.resources.get('1')).to.be.undefined()
       })
 
-      it('should request more subjects', function () {
-        while (rootStore.subjects.resources.size > 2) {
-          rootStore.subjects.advance()
-        }
-        // Once for initialization and once after the queue has been advanced to less than 3 subjects
-        expect(rootStore.subjects.populateQueue).to.have.been.calledTwice()
+      it('should reduce the queue size by one', function () {
+        const expectedSize = rootStore.subjects.resources.size - 1
+        rootStore.subjects.advance()
+        expect(rootStore.subjects.resources.size).to.equal(expectedSize)
```

Comment:
Tests are failing because `rootStore.subjects.resources.size` doesn't change here, and I can't immediately see why that would happen.

Reply:
Manually calling `populateQueue` fixed this. I don't know why.

Reply:
Because `populateQueue` is asynchronous. The call for it in the store instantiation's `afterAttach` wasn't done yet. This was resolved in other describe blocks by setting up the stores in a `before` or `beforeEach`. This test is now setting up the stores with an unfinished `populateQueue` call and then manually calls it for the expectations. 

It seems cleaner to me to do the store instantiation in a `before` or `beforeEach` since these are really integration tests now and we are testing the behavior of both actions here, but this may be more readable? I'm not sure what I'd prefer. What I'm concerned with is that people will read this and not realize we are calling `populateQueue` on store instantiation in its `afterAttach` lifecycle method.

Reply:
At this point, the code was using `before`. If you check out that commit, you should see that the tests fail.

Reply:
I’d much prefer to have this working with the initial setup in a `before` block. I noticed that the metadata tests had to use the same hack.

Reply:
I've removed that hack, because I think it makes it confusing to try and understand what the initial mock store state will be.

What you should see now is that adding a new expectation causes previously passing tests to fail. Do you have any ideas why the active subject is undefined after advancing the queue?

Reply:
This is passing now, so the latest commits are working.